### DISCUSSION TITLE
Does react native firebase support deep link campaigns?

### DISCUSSION THREAD

**Discussion by mercpls** [invertase/react-native-firebase#6369] [Q&A]
If we deep link to the store with UTM parameters, are we able to read these in the application on first open, or can we not do this with React Native firebase?

Does anyone know how the referrer information is passed from the app store to the app or if it is at all?

**Comment by mikehardy** [comment]
You might like https://rnfirebase.io/reference/dynamic-links#resolveLink it returns https://rnfirebase.io/reference/dynamic-links/dynamiclink and should have https://rnfirebase.io/reference/dynamic-links/dynamiclink#utmParameters

**Reply by mercpls** [reply]
We're already able to resolve links, the issue is specifically related to UTM campaigns and tracking if a user installed the app after clicking a link that contains a campaign. So we can reward affiliates who's users download our app.

**Reply by mikehardy** [reply]
You asked if you were able to read the UTM parameters
I posted that the object returned from resolveLink is documented as having them.

Are you actually having a problem? It *sounds* like you are using resolveLink (or getInitialLink?) but it's just not containing the UTM params?

If so, you are having an issue, but you started a discussion, which it makes it harder to help. You should file an actual issue filling in our template with an eye towards https://stackoverflow.com/help/how-to-ask / https://stackoverflow.com/help/mcve

**Reply by mercpls** [reply]
> You asked if you were able to read the UTM parameters I posted that the object returned from resolveLink is documented as having them.
> 
> Are you actually having a problem? It _sounds_ like you are using resolveLink (or getInitialLink?) but it's just not containing the UTM params?
> 
> If so, you are having an issue, but you started a discussion, which it makes it harder to help. You should file an actual issue filling in our template with an eye towards https://stackoverflow.com/help/how-to-ask / https://stackoverflow.com/help/mcve

The problem stems from being unable to get `initialLink` when the app is opened. It's a hard one to test, because testflight doesn't seem to send anything. And development builds also don't get one after clicking a deep link (which routes to the stores). I am not sure if the store is supposed to forward a deep link, or if google is finger printing, and then sending the deep link to the device.

**Reply by mikehardy** [reply]
I think to test a deep link from uninstalled state the standard process is to have a "clean" phone (no app installed), click on the deep link which should take you to app store, side-load the release build via Xcode and start it - this is very similar to something someone else was doing and it links in to a good firebase-ios-sdk issue explaining the process: check this and the link in it: https://github.com/invertase/react-native-firebase/discussions/6372#discussioncomment-3095050 - perhaps it helps?

**Reply by mercpls** [reply]
> I think to test a deep link from uninstalled state the standard process is to have a "clean" phone (no app installed), click on the deep link which should take you to app store, side-load the release build via Xcode and start it - this is very similar to something someone else was doing and it links in to a good firebase-ios-sdk issue explaining the process: check this and the link in it: [#6372 (comment)](https://github.com/invertase/react-native-firebase/discussions/6372#discussioncomment-3095050) - perhaps it helps?

Thank you so much! That’s a perfect link.
